src/api_manager/fetch_metadata.cc

```diff
@@ -136,15 +136,6 @@ void GlobalFetchServiceAccountToken(
       // the token fetch, while subsequent requests in the window reuse
       // the old token.
       break;
-    case auth::ServiceAccountToken::FETCHING:
-      env->LogDebug("Service account token fetch in progress");
-      // If token is still valid, continue
-      if (token->is_access_token_valid(0)) {
-        continuation(Status::OK);
-      } else {
-        continuation(Status(Code::UNAVAILABLE, kFetchingToken));
```

Comment:
How about even smaller change?

Just remove line 145 and flow continue to line 157.  we still keep FETCHING state

Reply:
Modified
      if (token->is_access_token_valid(0)) {
        continuation(Status::OK);
         return;
      }